src/System.Drawing.Primitives/src/System/Drawing/Color.cs

```diff
@@ -420,188 +420,119 @@ private long Value
             }
         }
 
+        private static void ThrowOutOfByteRange(int value, string name) =>
```

Comment:
What is the point of adding an extra method to throw the exception if it is only used by one method and it will introduce an extra line in the call stack?

Reply:
Moving that out allows `CheckByte()` to be inlined, and the JIT optimizes the call to the Throw method, as described in points 1 and 3a here: https://github.com/dotnet/corefx/blob/master/src/System.Memory/src/System/ThrowHelper.cs#L16-L20

That change is what makes `FromArgb(int, int, int, int)` 3x faster

Reply:
I see, nice, I thought it would've been inlined with the throw statement. Good to know.

Reply:
When `ThrowOutOfByteRange` is used only by one method, it could be written as local function, to tidy up the code a bit -- or are local function against conventions in corefx?

Reply:
We don't use them much - I think we just haven't thought about when the best time to use them is.

Reply:
Any objection to my using one here?  I think it's a good idea for this pattern

Reply:
I've seen them in a few places already but as @danmosemsft mentions we don't use them much yet, but I think this could be a good place to use it, as long as it doesn't hurt the perf gain and inlining we're looking for.